mikrotik/resource_interface_wireguard.go

```diff
@@ -0,0 +1,242 @@
+package mikrotik
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type interfaceWireguard struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &interfaceWireguard{}
+	_ resource.ResourceWithConfigure   = &interfaceWireguard{}
+	_ resource.ResourceWithImportState = &interfaceWireguard{}
+)
+
+// NewInterfaceWireguardResource is a helper function to simplify the provider implementation.
+func NewInterfaceWireguardResource() resource.Resource {
+	return &interfaceWireguard{}
+
+}
+
+func (i *interfaceWireguard) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	i.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (i *interfaceWireguard) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_interface_wireguard"
+}
+
+// Schema defines the schema for the resource.
+func (i *interfaceWireguard) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Creates a Mikrotik interface wireguard.",
```

Comment:
Add the explanation about Router OS v7+ here. This will be populated in the docs written by `tfplugindocs`.

Reply:
Do I just add "only supported by RouterOS v7+" at the end of the description or is there a different way to add it to let `tfplugindocs` populate it in the docs? I'm not too familiar with `tfplugindocs`.

Reply:
To circle back on this, I've generated the doc file with `tfplugindocs`